### PR TITLE
Set permanent UPnP lease when required

### LINKED ISSUE #14: Fallback to add an infinite UPnP mapping for routers that do not support permanent leases
Some routers, e.g. mikrotik, do not support temporary leases, but only permanent (with `MAPPING_LIFETIME=0`).

Network error is then
```
Hypertext Transfer Protocol
    HTTP/1.1 500 Internal Server Error\r\n
    CONTENT-LENGTH: 512\r\n
    CONTENT-TYPE: text/xml; charset="utf-8"\r\n
    DATE: Wed, 16 Jan 2019 11:55:17 GMT\r\n
    EXT: \r\n
    SERVER: RouterOS/6.34.4UPnP/1.0 MikroTik UPnP/1.0\r\n
    \r\n
    [HTTP response 1/1]
    [Time since request: 0.002875000 seconds]
    [Request in frame: 25394]
    File Data: 512 bytes
eXtensible Markup Language
    <s:Envelope
        xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"
        s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
        <s:Body>
            <s:Fault>
                <faultCode>
                    s:Client
                    </faultCode>
                <faultString>
                    UPnPError
                    </faultString>
                <detail>
                    <UPnPError
                        xmlns="urn:schemas-upnp-org:control-1-0">
                        <errorCode>
                            725
                            </errorCode>
                        <errorDescription>
                            OnlyPermanentLeasesSupported
                            </errorDescription>
                        </UPnPError>
                    </detail>
                </s:Fault>
            </s:Body>
        </s:Envelope>
```

The solution is to retry to add a mapping with `MAPPING_LIFETIME=0` when `OnlyPermanentLeasesSupported` error is returned by UPnP v1/v2.

Mind that `MAPPING_LIFETIME=0` is considered a permanent lease only by UPnP protocol, in NAT-PMP sending `MAPPING_LIFETIME=0` removes the mapping!

I would be nice to have fallback to infinite mapping configurable, e.g. by `allow_permanent_upnp_leases` option, to have it driven by user needs.

### PR DESCRIPTION
**Pull request by benoitc**
fix #14

### CODE REVIEW

File: src/natupnp_v1.erl [hunk 1]
```diff
@@ -169,9 +170,32 @@ add_port_mapping1(#nat_upnp{ip=Ip, service_url=Url}, Protocol, InternalPort,
             Now = nat_lib:timestamp(),
             MappingLifetime = Lifetime - (Now - Start),
             {ok, Now, InternalPort, ExternalPort, MappingLifetime};
```
Comment (benoitc) [inline src/natupnp_v1.erl]: what should return there if PermanentLease have been set? would returning `permanent_lease` or `infinity` be ok?

Comment (zp-sd) [reply src/natupnp_v1.erl]: yes, both seems fine 👍

**Comment by zp-sd** [comment]
@benoitc Should the changes be applied to `natupnp_v2.erl` as well? Also, can we make sending permanent lease configurable?

File: src/natupnp_v1.erl (hunk 1 above)
Comment (benoitc) [reply src/natupnp_v1.erl]: it will return infinity for permanent leases. Thanks for the feedback !

**Comment by benoitc** [comment]
@zp-sd now it is. Also tested on [amplifi](https://www.amplifi.com) router and it worked.

**Pull request merged**